operator/src/network/controller.rs

```diff
@@ -685,3 +700,141 @@ async fn apply_config_map(
         .await?;
     Ok(())
 }
+
+// Stub tests relying on stub.rs and its apiserver stubs
+#[cfg(test)]
+mod test {
+    use super::{reconcile, Context, Network};
+
+    use crate::network::{
+        stub::{timeout_after_1s, Stub},
+        utils::RpcClientMock,
+        NetworkSpec, NetworkStatus,
+    };
+
+    use expect_test::{expect, expect_file};
+    use keramik_common::peer_info::PeerInfo;
+
+    use std::sync::Arc;
+    use unimock::{matching, MockFn, Unimock};
+
+    // This tests defines the default stubs,
+    // meaning the default stubs are the request response pairs
+    // that occur when reconiling a default spec and status.
+    #[tokio::test]
+    async fn reconcile_from_empty() {
+        let mock_rpc_client = Unimock::new(());
+        let (testctx, fakeserver) = Context::test(mock_rpc_client);
+        let network = Network::test();
+        let mocksrv = fakeserver.run(Stub::default());
+        reconcile(Arc::new(network), testctx)
+            .await
+            .expect("reconciler");
+        timeout_after_1s(mocksrv).await;
+    }
+    #[tokio::test]
+    async fn reconcile_two_peers() {
+        // Setup network spec and status
+        let network = Network::test()
+            .with_spec(NetworkSpec {
+                replicas: 2,
+                ..Default::default()
+            })
+            .with_status(NetworkStatus {
+                replicas: 2,
+                ready_replicas: 0,
+                namespace: Some("keramik-test".to_owned()),
+                peers: vec![],
+            });
+        // Setup peer info
+        let mock_rpc_client = Unimock::new((
+            RpcClientMock::peer_info
+                .next_call(matching!(_))
+                .returns(Ok(PeerInfo {
+                    index: 0,
+                    peer_id: "peer_id_0".to_owned(),
+                    rpc_addr: "http://peer0".to_owned(),
+                    p2p_addrs: vec!["/ip4/10.0.0.1/tcp/4001/p2p/peer_id_0".to_owned()],
+                })),
+            RpcClientMock::peer_info
+                .next_call(matching!(_))
+                .returns(Ok(PeerInfo {
+                    index: 1,
+                    peer_id: "peer_id_1".to_owned(),
+                    rpc_addr: "http://peer1".to_owned(),
+                    p2p_addrs: vec!["/ip4/10.0.0.2/tcp/4001/p2p/peer_id_1".to_owned()],
+                })),
+        ));
```

Comment:
As an example here we mock out the RpcClient to return information about two peers in the network. Then the test validates the networks status contains that information.